radio_beam/tests/test_beam.py

```diff
@@ -83,36 +95,162 @@ def test_from_casa_image():
+
+    conv_factor = u.brightness_temperature(beam.sr, freq)
+
+    assert_quantity_allclose((1 * u.Jy).to(u.K, equivalencies=conv_factor),
+                             beam.jtok(freq))
+
+
+def test_jtok_equiv():
+
+    major = 0.1 * u.rad
+    beam = Beam(major, major, 30 * u.deg)
+
+    freq = 1.42 * u.GHz
+
+    conv_factor = u.brightness_temperature(beam.sr, freq)
+    conv_beam_factor = beam.jtok_equiv(freq)
+
+    assert_quantity_allclose((1 * u.Jy).to(u.K, equivalencies=conv_factor),
+                             (1 * u.Jy).to(u.K, equivalencies=conv_beam_factor))
+
+    assert_quantity_allclose((1 * u.K).to(u.Jy, equivalencies=conv_factor),
+                             (1 * u.K).to(u.Jy, equivalencies=conv_beam_factor))
+
+
+def test_convolution():
+
+    major1 = 1 * u.deg
+    minor1 = 0.5 * u.deg
+    pa1 = 0.0 * u.deg
+    beam1 = Beam(major1, minor1, pa1)
+
+    major2 = 1 * u.deg
+    minor2 = 0.75 * u.deg
+    pa2 = 90.0 * u.deg
+    beam2 = Beam(major2, minor2, pa2)
+
+    alpha = (major1 * np.cos(pa1))**2 + (minor1 * np.sin(pa1))**2 + \
+        (major2 * np.cos(pa2))**2 + (minor2 * np.sin(pa2))**2
+    beta = (major1 * np.sin(pa1))**2 + (minor1 * np.cos(pa1))**2 + \
+        (major2 * np.sin(pa2))**2 + (minor2 * np.cos(pa2))**2
+    gamma = 2 * ((minor1**2 - major1**2) * np.sin(pa1) * np.cos(pa1) +
+                 (minor2**2 - major2**2) * np.sin(pa2) * np.cos(pa2))
+
+    s = alpha + beta
+    t = np.sqrt((alpha - beta)**2 + gamma**2)
+
+    conv_major = np.sqrt(0.5 * (s + t))
+    conv_minor = np.sqrt(0.5 * (s - t))
+    conv_pa = 0.5 * np.arctan2(- gamma, alpha - beta)
+
+    conv_beam = beam1.convolve(beam2)
+
+    assert_quantity_allclose(conv_major, conv_beam.major)
+    assert_quantity_allclose(conv_minor, conv_beam.minor)
+    assert_quantity_allclose(conv_pa, conv_beam.pa)
+
+
+def test_deconvolution():
+
+    major1 = 2.0 * u.deg
+    minor1 = 1.0 * u.deg
+    pa1 = 45.0 * u.deg
+    beam1 = Beam(major1, minor1, pa1)
+
+    major2 = 1 * u.deg
+    minor2 = 0.5 * u.deg
+    pa2 = 0.0 * u.deg
+    beam2 = Beam(major2, minor2, pa2)
+
+    alpha = (major1 * np.cos(pa1))**2 + (minor1 * np.sin(pa1))**2 - \
+        (major2 * np.cos(pa2))**2 - (minor2 * np.sin(pa2))**2
+    beta = (major1 * np.sin(pa1))**2 + (minor1 * np.cos(pa1))**2 - \
+        (major2 * np.sin(pa2))**2 - (minor2 * np.cos(pa2))**2
+    gamma = 2 * ((minor1**2 - major1**2) * np.sin(pa1) * np.cos(pa1) +
+                 (minor2**2 - major2**2) * np.sin(pa2) * np.cos(pa2))
+
+    s = alpha + beta
+    t = np.sqrt((alpha - beta)**2 + gamma**2)
+
+    deconv_major = np.sqrt(0.5 * (s + t))
+    deconv_minor = np.sqrt(0.5 * (s - t))
+    deconv_pa = 0.5 * np.arctan2(- gamma, alpha - beta)
+
+    deconv_beam = beam1.deconvolve(beam2)
+
+    assert_quantity_allclose(deconv_major, deconv_beam.major)
+    assert_quantity_allclose(deconv_minor, deconv_beam.minor)
+    assert_quantity_allclose(deconv_pa, deconv_beam.pa)
+
+
+def test_conv_deconv():
```

Comment:
I meant for `failure_returns_pointlike=False` and `failure_returns_pointlike=True`.   Test that one raises exception, the other returns `Beam(0,0,0)`